handler/src/main/java/io/netty/handler/ssl/JdkAlpnSslEngine.java

```diff
@@ -62,12 +60,15 @@ private static void updateAvailability() {
                     "protocolSelector");
             ALPN.put(engine, new ServerProvider() {
                 @Override
-                public String select(List<String> protocols) {
+                public String select(List<String> protocols) throws SSLException {
                     try {
                         return protocolSelector.select(protocols);
+                    } catch(SSLException e) {
+                        throw e;
                     } catch (Throwable t) {
-                        PlatformDependent.throwException(t);
-                        return null;
+                        // Ensure that all exceptions are propagated as SSLExceptions
+                        // so that the SslHandler properly fails the handshake.
+                        throw new SSLException(t);
```

Comment:
Not a necessity but seems like it may be more specific to what is going on here.  We are intending to fail a handshake.

@ejona86 - You have been doing a bunch of investigation in this area...any thoughts?